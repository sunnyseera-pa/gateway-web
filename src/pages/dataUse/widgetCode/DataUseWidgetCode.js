/** @jsx jsx */
import { jsx } from '@emotion/react';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { OverlayTrigger, Tooltip } from 'react-bootstrap';
import PropTypes from 'prop-types';
import Typography from '../../../components/Typography';
import Button from '../../../components/Button';
import styles from './DataUseWidgetCode.styles';

const DataUseWidgetCode = ({ codeString, copyToClipBoard }) => {
    const { t } = useTranslation();
    const tooltip = <Tooltip id='tooltip'>{t('datause.widget.copyCodeClipBoard')}</Tooltip>;
    return (
        <div className='row'>
            <div className='col-12'>
                <Typography variant='h6' mb={3}>
                    {t('datause.widget.copyCodeHelp')}
                </Typography>{' '}
            </div>
            <div className='col-12'>
                <div css={styles}>
                    <pre>
                        <code>{codeString}</code>
                    </pre>
                </div>
            </div>
            <div className='col-12'>
                <OverlayTrigger placement='bottom' overlay={tooltip} trigger='click'>
                    <Button className='float-right' onClick={copyToClipBoard} type='button' mt={2}>
                        {t('datause.widget.copyCodeButton')}
                    </Button>
                </OverlayTrigger>
            </div>
        </div>
    );
};

DataUseWidgetCode.propTypes = {
    codeString: PropTypes.string.isRequired,
    copyToClipBoard: PropTypes.func.isRequired,
};

export default DataUseWidgetCode;
