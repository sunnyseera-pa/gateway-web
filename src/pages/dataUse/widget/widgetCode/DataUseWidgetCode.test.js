import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '@testing-library/react';
import DataUseWidgetCode from './DataUseWidgetCode';

let wrapper;
const copyToClipBoard = jest.fn();

describe('Given the DataUseWidgetCode component', () => {
    describe('When it is rendered', () => {
        beforeAll(() => {
            wrapper = render(<DataUseWidgetCode codeString='<test/>' copyToClipBoard={copyToClipBoard} />, {
                wrapper: Providers,
            });
        });

        it('Then matches the previous snapshot', () => {
            expect(wrapper.container).toMatchSnapshot();
        });

        it('Then  Copy Code button click should call correct function', () => {
            fireEvent.click(screen.getByText('Copy Code'));
            expect(copyToClipBoard.mock.calls.length).toBe(1);
        });
    });
});
