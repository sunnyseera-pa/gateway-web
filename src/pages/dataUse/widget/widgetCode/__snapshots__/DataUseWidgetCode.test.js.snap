// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the DataUseWidgetCode component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-12"
    >
      <h6
        class="ui-LayoutBox ui-Typography css-18e1tcr-Typography css-2lztm6"
      >
        Copy the source code and share with your development team to display the widget above on your website
      </h6>
       
    </div>
    <div
      class="col-12"
    >
      <pre
        style="display: block; overflow-x: auto; padding: 0.5em; background: rgb(240, 240, 240); color: rgb(68, 68, 68);"
      >
        <code
          class="language-html"
          style="white-space: pre;"
        >
          <span
            style="color: rgb(136, 0, 0); font-weight: bold;"
          >
            &lt;test/&gt;
          </span>
        </code>
      </pre>
    </div>
    <div
      class="col-12"
    >
      <button
        class="ui-LayoutBox float-right css-kp49dp-Button css-1k6rz1e"
        type="button"
      >
        Copy Code
      </button>
    </div>
  </div>
</div>
`;
